### PR TITLE
SI-8035 Deprecate automatic () insertion in argument lists

### PR DESCRIPTION
**Pull request by soc**
This promotes the () insertion warning from -Ywarn-adapted-args to a
deprecation warning. -Xfuture tunrs it into a compiler error.

Auto tupling remains unchanged for now.

The tests have been fixed the following way:
- Warnings caused by general sloppiness (Try(), Future(), ...) have been
  fixed.
- Warnings which raise interesting questions (x == (), ...) received an
  updated checkfile for now.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Adaptations.scala [hunk 1]
```diff
@@ -69,17 +69,20 @@ trait Adaptations {
...
-          if (args.isEmpty) "Adapting argument list by inserting (): " + (
-            if (isLeakyTarget) "leaky (Object-receiving) target makes this especially dangerous."
-            else "this is unlikely to be what you want."
-          )
-          else "Adapting argument list by creating a " + args.size + "-tuple: this may not be what you want."
-        )
+        context.warning(t.pos, adaptWarningMessage("No automatic adaptation here: use explicit parentheses."))
+      else if (args.isEmpty) {
+        if (settings.future)
+          context.error(t.pos, adaptWarningMessage("Adaption of argument list by inserting () has been removed.", showAdaption = false))
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Adaptations.scala]: s/Adaption/Adaptation/

**Comment by soc** [comment]
@retronym Fixed.

**Comment by retronym** [comment]
Could you please summarize the change in the commit message? (e.g. "it used to incur a warning under -Y...; now we promote the case of ()-insertion to a deprecation (or an error under -Xfuture); tupling of other arities remains unchanged for now.")

**Comment by soc** [comment]
@retronym Done.

**Comment by soc** [comment]
Review by @retronym, please!

**Comment by xeno-by** [comment]
@soc Needs a rebase

**Comment by soc** [comment]
@xeno-by Done.

**Comment by adriaanm** [comment]
LGTM

**Comment by sief** [comment]
I get the deprecation warning in the following snippet:

scala> def f(f1: Unit => Unit) {f1()}
<console>:7: warning: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.

If I omit the parentheses I get a different warning which makes sense, I need the parentheses in order to execute f1. Am I missing something here?

**Comment by retronym** [comment]
```
scala> def f(f1: Unit => Unit) {f1()}
<console>:7: warning: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
        signature: Function1.apply(v1: T1): R
  given arguments: <none>
 after adaptation: Function1((): Unit)
       def f(f1: Unit => Unit) {f1()}
                                  ^
f: (f1: Unit => Unit)Unit

scala> def f(f1: Unit => Unit) {f1(())}
f: (f1: Unit => Unit)Unit
```

**Pull request merged**